runtime/util/rcdump.c

```diff
@@ -823,6 +829,22 @@ dumpEnclosingMethod(J9PortLibrary *portLib, J9ROMClass *romClass, U_32 flags)
 	return BCT_ERR_NO_ERROR;
 }
 
+static I_32
+dumpPermittedSubclasses(J9PortLibrary *portLib, J9ROMClass *romClass)
+{
+	PORT_ACCESS_FROM_PORT(portLib);
+
+	U_32 *permittedSubclassesCountPtr = getNumberOfPermittedSubclassesPtr(romClass);
```

Comment:
`getNumberOfPermittedSubclassesPtr` seems a strange API to me: why return a pointer (nobody appears to write to it)?
If we decide to change it, that change should be deferred to a separate pull request.

Reply:
Yes its a bit odd to the other optinfo api methods. Other variable-length sections of ROM class optionalInfo have iterator like setups but that seemed like overkill to me here since its only a list of class names that is stored.

The list of permitted subclass names is located directly after `permittedSubclassesCountPtr` so `permittedSubclassesNameAtIndex` takes in this pointer later as a starting address to find the class name.